### PR TITLE
Mkbhaskar/issue191

### PR DESCRIPTION
**Pull request by mkbhaskar**
Transitioned `launch_control.py` away from `while True: gui.force_update()` loop structure by moving log streaming process to a separate thread, and running the GUI execution loop instead (`app.exec_()`)

Similarly, to get the proxy to work, rather than running `while True: log_controller.update_proxy()`, switched proxy updating to a separate thread which only refreshes every 100 ms. This is sufficient for near-seamless updating of the proxy log console and lag-free operation of the GUI (which runs in the `app.exec_()` thread).

Checked CPU usage in Linux and Windows and verified near negligible idle CPU usage, as compared to constant 100% thread usage in the previous version.

Notes: 
1. will need to enforce `app.exec_()` implementation in all additional scripts that are not yet using this
2. the example of multi-threading and pyqt signals/slots in `launch_control.py` can be roughly coupied in case an application needs to run an intensive process which interfaces with a GUI running in its own `app.exec_()` thread.
3. each proxy launch control now launches its own GUI server, primarily for proper closing of the proxy launch control when the master is closed (no hanging processes remaining)

### CODE REVIEW

File: pylabnet/launchers/launch_control.py [hunk 1]
```diff
@@ -826,54 +894,42 @@ def main_staticproxy():
...
+            log_controller.start_stop_logging(master_log=True)
 
-    # Exit, close servers if necessary
-    if log_controller.proxy:
-        pass
-    else:
-        log_controller.kill_servers()
+        # Start thread to listen for updates
+        updater = UpdateReceiver(queue)
+        updater.update_signal.connect(log_controller.update)
```
Comment (cknaut) [inline pylabnet/launchers/launch_control.py:926]: @mkbhaskar , do I understand it correctly that the `UpdateReceiver` is launched in a new thread, which runs continuously and can communicate with the parent thread by emitting a signal, which you can then connect to a function defined in the parent thread?

Comment (mkbhaskar) [reply pylabnet/launchers/launch_control.py:926]: yeah, the reason being that external threads themselves are not allowed to directly call gui functions. So instead you need to use signals/slots to communicate between threads.

the alternative of course is to do all the processing in the main gui thread, but then you make a tradeoff between performance of the rapidly refreshing task and the gui responsiveness. this way, the gui can run as desired in its `app.exec_()` loop and you can independently control the rate at which the update process occurs

Comment (cknaut) [reply pylabnet/launchers/launch_control.py:926]: Ok, I think your implementation makes sense. Let's try it together with our new master-logger infrastructure and monitor performance.

**Review by cknaut — APPROVED** [review]
Looks good.

**Pull request merged**